resources/build/build-utils.sh

```diff
@@ -137,3 +137,39 @@ findTier
 findVersion
 # printVersionUtilsDebug
 printBuildNumberForTeamCity
+
+# Intended for use with macOS-based builds, as Xcode build phase "run script"s do not have access to important
+# environment variables.  Doesn't hurt to run it at other times as well.  The output file is .gitignore'd.
+function exportEnvironmentDefinitionScript() {
+    ENVIRONMENT_SH="$KEYMAN_ROOT/resources/environment.sh"
+
+    # Remove old copy if it exists
+    rm $ENVIRONMENT_SH > /dev/null 2> /dev/null 
+
+    # Documentation about the script, within the script.
+    echo "# Do not edit - this is an autogenerated script.  See build/build-utils.sh for more details." >> $ENVIRONMENT_SH
```

Comment:
Please ensure `$ENVIRONMENT_SH` variable is quoted -- if there are spaces in the path, it'll break without that